### ISSUE TITLE
false positive "should be declared external" on abstract contract

### ISSUE THREAD

**Issue by thedavidmeister** [crytic/slither#883]
if a contract is marked `abstract` we require it be inherited so `public` functions do not need to be `external` if a child contract might reasonably want to use this function

**Comment by montyly** [comment]
Hi @thedavidmeister.

Thank you for reporting this. 

If the abstract contract is inherited, and the function is used somewhere, slither will not warm about the visibility, such as in:

```solidity
abstract contract A{
    function f() public virtual;
}

contract B is A{
    function f() public override{
    }
    function test() external {
        f();
    } 
}
```

Do you refer to abstract contracts that are never inherited? If so could you give more insight into their usages?

**Comment by thedavidmeister** [comment]
yes abstract contracts that are not inherited in the scope of the scan (but might be elsewhere, such as an open source utility)
